File: frontend/types/client_issue.ts
import {Attachment} from "./issue";

export interface ClientIssue{
    id: number;
    title: string;
    description?: string;
    status: string; //"open" | "in_progress" | "closed"
    priority: string; // "low" | "medium" | "high"
    created_at: string;
    updated_at?: string;
    attachments: Attachment[];
}

export interface ClientIssueCreatePayload {
    title: string;
    description?: string;
    priority: string; // "low" | "medium" | "high"
}

File: backend/app/schemas/client_issue.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from app.schemas.issue import IssueAttachmentOut

class ClientIssueCreate(BaseModel):
    title: str
    description: Optional[str] = None
    priority: Optional[str] = "medium"

class ClientIssueRead(BaseModel):
    id: int
    title: str
    description: Optional[str] = None
    status: str
    priority: str
    created_at: datetime
    updated_at: Optional[datetime] = None
    attachments: List[IssueAttachmentOut] = []

    class Config: 
        from_attributes = True
        